src/amp-viewer-host.js

```diff
@@ -22,5 +22,139 @@ import {
 } from 'amp-viewer-messaging/messaging';
 
 
+const CHANNEL_OPEN_MSG = 'channelOpen';
+
 export class AmpViewerHost {
+
+  /**
+   * @param {!Window} win
+   * @param {!HTMLIFrameElement} ampIframe
+   * @param {string} frameOrigin
+   * @param {function(string, *, boolean):(!Promise<*>|undefined)} messageHandler
+   * @param {boolean=} opt_isWebview Should viewer initiate handshake w/ polling
+   * @param {boolean=} opt_isHandshakePoll
+   * looking at.
+   */
+  constructor(win, ampIframe, frameOrigin, messageHandler, opt_isWebview,
+    opt_isHandshakePoll) {
+    /** @const {!Window} */
+    this.win = win;
+    /** @private {!HTMLIFrameElement} */
+    this.ampIframe_ = ampIframe;
+    /** @private {function(string, *, boolean):(!Promise<*>|undefined)} */
+    this.messageHandler_ = messageHandler;
+    /** @const {boolean} */
+    this.isWebview_ = !!opt_isWebview;
+
+    if (this.isWebview_ || opt_isHandshakePoll) {
```

Comment:
For the prod code, the constructor connections are not desirable usually. Hard to test and generally hard to customize/use.